src/codeql.test.ts

```diff
@@ -21,376 +20,414 @@ import {
 } from "./codeql";
 import { BaseCodeqlCli, CodeqlCli } from "./codeql-cli";
 
-const test = anyTest as TestFn<{
-  db: string;
-  tmpDir: string;
-  dbTmpDir: string;
-  codeql: CodeqlCli;
-}>;
-
-test.before(async (t) => {
-  const dbTmpDir = path.resolve(fs.mkdtempSync(path.join(tmpdir(), "db-")));
-  t.context.dbTmpDir = dbTmpDir;
-
-  const projectDir = path.join(dbTmpDir, "project");
-  const dbDir = path.join(dbTmpDir, "db");
-  fs.mkdirSync(projectDir);
-  const testFile = path.join(projectDir, "test.js");
-  fs.writeFileSync(testFile, "const x = 1;");
-
-  const codeql = new BaseCodeqlCli(process.env.CODEQL_BIN_PATH || "codeql");
-
-  await codeql.run([
-    "database",
-    "create",
-    "--language=javascript",
-    `--source-root=${projectDir}`,
-    dbDir,
-  ]);
-
-  const dbZip = path.join(dbTmpDir, "database.zip");
-  await codeql.run(["database", "bundle", `--output=${dbZip}`, dbDir]);
-  t.context.db = dbZip;
-
-  t.context.codeql = codeql;
-});
-
-test.after(async (t) => {
-  if (t.context?.dbTmpDir) {
-    await rmRF(t.context.dbTmpDir);
-  }
-});
-
-test.beforeEach((t) => {
-  // Use a different temporary directory that tests can use
-  t.context.tmpDir = path.resolve(fs.mkdtempSync(path.join(tmpdir(), "tmp-")));
-});
-
-test.afterEach(async (t) => {
-  if (t.context?.tmpDir !== undefined) {
-    await rmRF(t.context.tmpDir);
-  }
-});
-
-test("getting query pack info", async (t) => {
-  const queryPackInfo = await getQueryPackInfo(
-    t.context.codeql,
-    "testdata/test_pack",
+describe("codeql", () => {
+  const codeql: CodeqlCli = new BaseCodeqlCli(
+    process.env.CODEQL_BIN_PATH || "codeql",
   );
```

Comment:
Should we be consistent in when we create this object? In the `codeql-cli.test.ts` we create this in the `beforeAll` hook, while here we create it in the `describe` block. I have a preference for doing this in the `beforeAll` hook, but I'll leave it up to you which option you think is best.

Reply:
I've moved this to be in `beforeAll` so we're more consistent and avoiding re-using objects between tests.